shielder/README.md

```diff
@@ -0,0 +1,136 @@
+```
+Which yields:
+```
+assets=[Asset { token_id: 0, token_amount: 35, deposit_id: 0 }]
+```
+(note how the notes have been merged into one).
+
+And later as Hans (`//1`):
+```bash
+./target/release/shielder-cli deposit 0 33
+```
+
+Which in turn yields:
+```
+assets=[Asset { token_id: 0, token_amount: 68, deposit_id: 0 }]
+```
+
+An interesting takeaway from those trivial experiments is that even by inspecting what's inside the Shielder, we can't really say anything useful about the state of ownership.
+Of course, you can try depositing the second token (1) and see a second entry created in the assets lists.
+
+A natural course of action after checking out the deposists would be to check out the withdrawals. Let's do just that!
+
+### Interacting with PSP22 contracts
+
+Before we proceed with our withdrawals, let's figure out a way to inspect the balances of our PSP22 tokens. There are two ways of doing that: the Contracts UI and the `cargo contracts` CLI. We will use the former on account of the simplicity and user-friendliness.
+First, navigate to https://contracts-ui.substrate.io/ and in the top-left corner choose 'Local node'.
+![Screenshot from 2023-03-29 00-50-28](https://user-images.githubusercontent.com/3109645/228384333-ae302d6e-bd58-47e1-b25c-dcc93c658751.png)
+
+We will need to add our contracts to the UI to be able to interact with them. We will focus on the first contract: adding the second one follows the exact same pattern.
+First, let's choose 'Add New Contract' in the top-left corner. Then, when shown a new screen, we will need to choose 'Use On-Chain Contract Address'.
+
+You can learn the contract addresses from the `deploy/addresses.json` file. Let's use `"token_a_address"`, or: `5Ct2Gc8hQscGdhxtso5DP3EWRgk6nww733CpKCiA3qdZ2T8u` (make sure to inspect the file for yourself).
+You will need to paste the address in the 'Contract Address' field. Once you do that, two more fields will show up. You can choose any name of your liking for the 'Contract Name'. For the metadata, however, you will need to upload the `public_token/target/ink/public_token.json` file.
+After completing all that and clicking 'Add contract', you will be able to interact with the PSP22 contract.
+
+To call the contract as either Damian or Hans, you will need to click the `Caller` field and, despite its lack of responsiveness, paste the appropriate address there. Let's use `5D34dL5prEUaGNQtPPZ3yN5Y6BnkfXunKXXz6fo7ZJbLwRRH` (Damian) for the sake of this tutorial.
+
+This will allow you to easily check your balance by selecting the `balanceOf` method and again pasting Damian's address. This way we can check the balance before and after the withdrawal from Shielder and be extra sure the the funds actually ended up in the right account.
+
+### Withdrawing assets
+
+Withdrawing assets is in every sense dual to depositing them. We will run a command and see the funds appear back in our balances of the corresponding PSP22 tokens.
```

Comment:
`in every sense dual to depositing them -> in every sense a mirror image of depositing them`

Reply:
complementary?